dotnet/src/SemanticKernel.Core/PromptTemplate/KernelPromptTemplate.cs

```diff
@@ -136,5 +139,29 @@ internal IList<Block> RenderVariables(IList<Block> blocks, KernelArguments? argu
             ? block
             : new TextBlock(((ITextRendering)block).Render(arguments), this._loggerFactory)).ToList();
     }
+
+    /// <summary>
+    /// Validates that all the KernelArguments are of string type.
+    /// </summary>
+    /// <param name="arguments">The collection of KernelArguments to validate.</param>
+    /// <exception cref="KernelException">Thrown when an argument is not of type string.</exception>
+    private static void AssertArgumentOfStringType(KernelArguments? arguments)
+    {
+        if (arguments == null)
+        {
+            return;
+        }
+
+        foreach (var argument in arguments)
+        {
+            if (argument.Value is string)
+            {
+                continue;
+            }
+
+            throw new KernelException($"Kernel prompt template arguments must be of type string. Argument '{argument.Key}' is of type '{argument.Value?.GetType()}'");
```

Comment:
Let's explain in the message (and the others) that this is a temporary state. 

E.g. `Non string kernel prompt template arguments are not supported in release candidate 1. This feature is coming soon but for now please make all arguments strings.`

@matthewbolanos Can you help here with the exact message to use?

Would this be a better exception: https://learn.microsoft.com/en-us/dotnet/api/system.invalidoperationexception?view=net-8.0